protected/humhub/components/InstallationState.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace humhub\components;
+
+use humhub\helpers\DatabaseHelper;
+use humhub\libs\DatabaseCredConfig;
+use Yii;
+use yii\base\BaseObject;
+use yii\base\StaticInstanceInterface;
+use yii\base\StaticInstanceTrait;
+
+class InstallationState extends BaseObject implements StaticInstanceInterface
+{
+    use StaticInstanceTrait;
+
+    public const STATE_NOT_INSTALLED = 0;
```

Comment:
Would be great to have a small description to each states. To better understand.

Then we should revise the states. e.g.
e.g. do we really need in the "ModuleManager" that many State checks and why? Probably it was required, to allow Modules during the Installer process.